Source/WebCore/Modules/cookie-store/CookieStore.cpp

```diff
@@ -254,6 +254,27 @@ void CookieStore::get(CookieStoreGetOptions&& options, Ref<DeferredPromise>&& pr
         return;
     }
 
+    if (options.name.isNull() && options.url.isNull()) {
+        promise->reject(ExceptionCode::TypeError);
+        return;
+    }
+
+    auto url = context->url();
+    if (!options.url.isNull()) {
+        auto parsed = context->completeURL(options.url);
+        if (context->isDocument() && parsed != url) {
+            promise->reject(ExceptionCode::TypeError);
+            return;
+        }
+
+        if (!origin->isSameOriginAs(SecurityOrigin::create(parsed))) {
+            promise->reject(ExceptionCode::TypeError);
+            return;
+        }
+        url = WTFMove(parsed);
+    }
+    options.url = url.string();
```

Comment:
Diving deeper at this code, I think `options.url` is never used in our existing implementation (e.g. in `NetworkStorageSession::cookiesForSessionAsVector` only `options.name` is checked). Please double check to make sure it is used correctly. It can be fixed in a separate patch.

Reply:
That's a good point. The code currently uses document's url for the url in `NetworkStorageSession::cookiesForSessionAsVector` but should instead use this error-checked URL that this patch adds. We should likely pass it in as a parameter the way `CookieStore::getAll()` does rather than passing it through `options.url`. I will look into this as part of the next patch.